server/server.go

```diff
@@ -156,6 +158,23 @@ func (s *Server) ServeDebug(l net.Listener) error {
 // Stop the containerd server canceling any open connections
 func (s *Server) Stop() {
 	s.rpc.Stop()
+	for i := len(s.plugins) - 1; i >= 0; i-- {
+		p := s.plugins[i]
+		instance, err := p.Instance()
+		if err != nil {
+			log.L.WithError(err).
+				Errorf("could not get plugin instance %q", p.Registration.ID)
+			continue
+		}
+		closer, ok := instance.(plugin.Closer)
+		if !ok {
+			continue
+		}
+		if err := closer.Close(); err != nil {
+			log.L.WithError(err).
+				Errorf("failed to close plugin %q", p.Registration.ID)
```

Comment:
Use `WithField` here for the registration id.

Reply:
Will do.